infrastructure/webdriver/bidi/subscription.html

```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Test console log are present</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script>
+    promise_test(async () => {
+        const some_message = "SOME MESSAGE";
+        // Subscribe to `log.entryAdded` BiDi events. This will not add a listener to the page.
+        await test_driver.bidi.log.entry_added.subscribe();
+        // Add a listener for the log.entryAdded event. This will not subscribe to the event, so the subscription is
+        // required before. The cleanup is done automatically after the test is finished.
+        const log_entry_promise = new Promise(resolve => test_driver.bidi.log.entry_added.on(resolve));
+        // Emit a console.log message.
+        console.log(some_message)
```

Comment:
Looks like the linter complains about this line. Also the infrastructure test seems to be currently failing.

Reply:
The infra test is expected to fail with marionette, as BiDi is not implemented there. IDK if we want to add expectations or something else.

```
Traceback (most recent call last):
  File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/executormarionette.py", line 898, in run_func
    self.result = True, self.func(self.protocol, self.url, self.timeout)
  File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/executormarionette.py", line 1024, in do_testharness
    done, rv = handler(result)
  File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/base.py", line 744, in __call__
    return callback(url, payload)
  File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/base.py", line 757, in process_action
    raise ValueError(f"Unknown action {action}") from e
ValueError: Unknown action bidi.session.subscribe
```

Reply:
Should the error happen in testdriver-vendor.js that the BiDi methods are not implemented? E.g, https://github.com/web-platform-tests/wpt/pull/45823/files#diff-1fe2b624679a3150e5c86f84682c5901b715dad750096a524e8cb23939e5590fR1124

Reply:
> Should the error happen in testdriver-vendor.js that the BiDi methods are not implemented? E.g, https://github.com/web-platform-tests/wpt/pull/45823/files#diff-1fe2b624679a3150e5c86f84682c5901b715dad750096a524e8cb23939e5590fR1124

`wptrunner` implements [`window.test_driver_internal.bidi....`](https://github.com/web-platform-tests/wpt/blob/14e4b58874dfc283205e89cbab6f08db8d57be3d/tools/wptrunner/wptrunner/testdriver-extra.js#L199) methods. Other vendor-specific transports can have it implemented or not, and the exception would depend on it. In case of `wptrunner` (`WebDriverTestharnessExecutor`), the specific protocol part can be present or not, but the protocol state is not exposed to the testdriver.

Reply:
Looks like the infrastructure still fails (could be unrelated to the changes here).

Reply:
Any other objections except the CI failures?

Reply:
The failing infra test for **`firefox_android`**:
> 2:41.20 TEST_START: /infrastructure/assumptions/document-fonts-ready.html
TEST-START | /infrastructure/assumptions/document-fonts-ready.html
 2:41.22 INFO Closing window 5c02c62a-826d-493a-ae98-35a8dc934466
Closing window 5c02c62a-826d-493a-ae98-35a8dc934466
mem avail: 12462 of 15806 MiB (78 %), swap free:    0 of    0 MiB ( 0 %)
 3:36.20 INFO Got timeout in harness
Got timeout in harness
 3:36.29 TEST_END: TIMEOUT, expected OK - TestRunner hit external timeout (this may indicate a hang)

Reply:
Experimental infra test PR: https://github.com/web-platform-tests/wpt/pull/46788

Reply:
So this time the tests have passed

Reply:
I consider this concern is addressed